src/murmur/ServerDB.cpp

```diff
@@ -1707,7 +1707,7 @@ void Server::updateChannel(const Channel *c) {
 				qFatal("ServerDB: internal query failure: PostgreSQL query did not return the inserted group's group_id");
 			}
 		} else {
-			SQLPREP("INSERT INTO `%1groups` (`server_id`, `channel_id`, `name`, `inherit`, `inheritable`) VALUES (?,?,?,?,?)");
+			SQLPREP("INSERT INTO `%1groups` (`server_id`, `channel_id`, `name`, `inherit`, `inheritable`) VALUES (?,?,?,?,?) ON DUPLICATE KEY UPDATE server_id = server_id");
```

Comment:
All values should be updated in case of a conflict and not just the `server_id`. Or are there specific reasons not to that I missed?

Reply:
Yes i think it is a good idea. 
Sorry i'm not DEV and i'm not used to using github, do you have the possibility to do changes please ?

Reply:
Yes I can do that. Not immediately though but I'll assign myself to this and will do so once I have the time :)

Reply:
Of course ! Thanks !

Reply:
No problem :grinning:

Reply:
Okay I just dug a bit deeper into this and `ON DUPLICATE KEY UPDATE` is a MySQL specific extension that is not contained in the SQL standard. And the code part you have changed is not MySQL specific (e.g. on my machine this runs with SQLite).

Reply:
I think what we need here is to introduce a check on whether an entry with the given ID exists. If so, we want to use `UPDATE` and if not, we want to `INSERT`.
I'm really not good with SQL though, so if you know something about how this could be done, please let me know :)

/cc @davidebeatrici